docs/guides/typescript.md

```diff
@@ -76,4 +76,55 @@ Remix has TypeScript type definitions built-in as well. The starter templates cr
 
 <docs-info>Note that the types referenced in `remix.env.d.ts` will depend on which environment you're running your app in. For example, there are different globals available in Cloudflare</docs-info>
 
+## Type Utilities
+
+Remix's built-in inference between `loader` function + `useLoaderData` generic and the `action` function + `useActionData` generic is quite good and you shouldn't often have to use these utilities. That said they can be quite handy in some cases which is why they are exported.
+
+### `UseDataFunctionReturn`
+
+Use this to get the type you'll get out of a `useLoaderData` or `useActionData` (basically, what the client will get as a result of what the `loader` or `action` function return):
+
+```tsx lines=[31]
+import { getCats } from "~/models/cats.server";
+
+export const loader: LoaderFunction = () => {
+  return json({ cats: await getCats() });
+};
+
+export const action: ActionFunction = () => {
+  return json({ error: "Not yet implemented" });
+};
+
+export default function CatsRoute() {
+  const data = useLoaderData<typeof loader>();
```

Comment:
I did some local testing and I think the type of `data` will be `any`. Changing to `ActionArgs` and `LoaderArgs` instead of setting type for the entire function does the trick.

```
export const action = async ({ request }: ActionArgs) => {
};
```

Reply:
Just got redirected to this PR by @kentcdodds to check some examples around typings and figured I'd give some feedback! If this is WIP, please consider my comment irrelevant 👋